packages/ckeditor5-dev-tests/bin/testautomated.js

```diff
@@ -13,6 +13,14 @@ const tests = require( '../lib/index' );
 
 const options = tests.parseArguments( process.argv.slice( 2 ) );
 
+// We want to mark tests as failed if any warnings are logged. See https://github.com/cksource/ckeditor5-internal/issues/3299
+if ( options.production ) {
+	console.warn = errors => {
+		console.log( errors );
```

Comment:
I tried calling `console.error` instead, but it does not break the execution.
The question is should we also throw error on `console.error`?

Reply:
I think that both `console.warn` and `console.error` should break the execution and throw an error.

Reply:
During a call, we created a custom plugin. Could you guys verify if it works on your machines too?

Webpack config for automated tests:

```js
if ( options.production ) {
	config.plugins.push(
		new class TreatWarningsAsErrorsWebpackPlugin {
			apply( compiler ) {
				compiler.hooks.shouldEmit.tap( 'TreatWarningsAsErrorsWebpackPlugin', compilation => {
					const stats = compilation.getStats();

					compilation.errors = [
						...compilation.errors,
						...compilation.warnings
					];

					compilation.warnings = [];

					if ( compilation.errors.length ) {
						return false;
					}
				} );
			}
		}
	)
}
```

Reply:
It works. Test run is terminated.